test/python/common.py

```diff
@@ -308,42 +356,20 @@ def requires_qe_access(func):
     """
 
     @functools.wraps(func)
-    def _wrapper(*args, **kwargs):
-        if SKIP_ONLINE_TESTS:
+    def _wrapper(self, *args, **kwargs):
+        if TEST_OPTIONS['skip_online']:
             raise unittest.SkipTest('Skipping online tests')
 
         # Cleanup the credentials, as this file is shared by the tests.
         from qiskit.wrapper import _wrapper
         _wrapper._DEFAULT_PROVIDER = DefaultQISKitProvider()
+        kwargs.update(_get_credentials(self, TEST_OPTIONS))
 
-        if os.getenv('USE_ALTERNATE_ENV_CREDENTIALS', False):
-            # Special case: instead of using the standard credentials mechanism,
-            # load them from different environment variables. This assumes they
-            # will always be in place, as is used by the Travis setup.
-            kwargs.update({
-                'qe_token': os.getenv('IBMQ_TOKEN'),
-                'qe_url': os.getenv('IBMQ_URL')
-            })
-            args[0].using_ibmq_credentials = True
+        if TEST_OPTIONS['rec'] or TEST_OPTIONS['mock_online']:
+            wrapper = VCR.use_cassette()(func)
```

Comment:
Replace with `func = VCR.use_cassette()(func)` and remove the `else` branch.

Reply:
It seems counterintuitive we `use_casette` while recording, but it is not our fault if they chose that API, can you clarify in a comment that, when using `rec`, that implies `VCR` is in recording mode?

Reply:
The statement `func = VCR.use_cassette()(func)` creates a funny situation in the scope static detection for lint. The error `Using variable 'func' before assignment (used-before-assignment)` pops up. Shall I ignore it?

In the way I see, you still need a cassette to record in a real-life VCR. Comment in: 71a8cb6